apps/dashboard/src/components/workflow-editor/steps/email/variables/variables.ts

```diff
@@ -1,49 +1,170 @@
+  const variable: LiquidVariable = { name: queryWithoutSuffix, aliasFor };
 
-  editor?.commands.deleteRange({ from, to });
-  editor?.commands.insertContent({
-    type: 'variable',
-    attrs: {
+  if (!isAllowedVariable(variable)) return;
+
+  // Calculate range for manual typing if not provided by suggestion
+  const calculatedRange = range || {
+    from: editor.state.selection.from - queryWithoutSuffix.length - 4, // -4 for '{{ }}'
+    to: editor.state.selection.from,
+  };
+
+  insertNodeToEditor({
+    editor,
+    range: calculatedRange,
+    nodeType: 'variable',
+    nodeAttrs: {
       id: queryWithoutSuffix,
+      aliasFor,
       label: null,
       fallback: null,
       showIfKey: null,
       required: false,
     },
   });
-}
+};
+
+const getVariablesByContext = (
+  editor: TiptapEditor,
+  from: VariableFrom,
+  isEnhancedDigestEnabled: boolean,
+  primitives: Array<LiquidVariable>,
+  arrays: Array<LiquidVariable>,
+  namespaces: Array<LiquidVariable>
+): LiquidVariable[] => {
+  const iterables = [...arrays, ...getRepeatBlockEachVariables(editor)];
+  const isInRepeatBlock = isInsideRepeatBlock(editor);
+
+  switch (from) {
+    // Case 1: Inside repeat block's "each" key input - only allow iterables
+    case VariableFrom.RepeatEachKey:
+      if (isInRepeatBlock) {
+        updateRepeatBlockChildAliases(editor, isEnhancedDigestEnabled);
+        return iterables;
+      }
+
+      return [];
+
+    // Case 2: Bubble menu (showIf) - allow only primitives and namespaces
+    case VariableFrom.Bubble:
+      return [...primitives, ...namespaces];
+
+    // Case 3: Regular content
+    case VariableFrom.Content: {
+      const baseVariables = [...primitives, ...namespaces, ...iterables];
+
+      // If we're not in a repeat block, return all variables
+      if (!isInRepeatBlock || !isEnhancedDigestEnabled) {
+        return baseVariables;
+      }
+
+      // If we're in a repeat block, return only the iterable properties (current + children)
+      const iterableName = editor?.getAttributes('repeat')?.each;
+      if (!iterableName) return baseVariables;
+
+      // Get all variables that are children of the iterable/alias
+      const iterableProperties = [...namespaces, ...arrays, ...primitives]
+        .filter((variable) => variable.name.startsWith(iterableName))
+        .flatMap((variable) => {
+          // If the variable name is exactly the iterableName, return just the alias
+          if (variable.name === iterableName) {
+            return [{ name: REPEAT_BLOCK_ITERABLE_ALIAS }];
+          }
+
+          // Otherwise, get the last part after the iterableName
+          const suffix = variable.name.split('.').pop();
+          return suffix ? [{ name: `${REPEAT_BLOCK_ITERABLE_ALIAS}.${suffix}` }] : [];
+        });
+
+      // Return all variables, including the iterable alias and its properties
+      return [...baseVariables, ...iterableProperties];
```

Comment:
Support for having available props after alias:

![image](https://github.com/user-attachments/assets/e1bf4697-36dd-499c-89db-20a80c531e9a)

Reply:
I also reworked the calculateVariables function again to be a bit more simpler